maven-plugin-tests/pom.xml

```diff
@@ -0,0 +1,54 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.apicurio</groupId>
+        <artifactId>apicurio-codegen-parent</artifactId>
+        <version>1.0.12-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>apicurio-codegen-maven-plugin-tests</artifactId>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>basic-codegen-test</module>
+        <module>widget-api-test</module>
+    </modules>
+
```

Comment:
To avoid issues with caches etc. it might be worth using the `maven-invoker` plugin:
https://github.com/fabric8io/kubernetes-client/blob/e520e0dd84b2a7b94d4f0bc8ee72a02bd056172d/java-generator/it/pom.xml#L69-L110

Reply:
Sorry I'm not sure I follow.  What does that solve?

Reply:
Using the `maven-invoker` is the "standard" way to test Maven Plugins AFAIK.
From the docs: https://maven.apache.org/shared/maven-invoker/index.html
I particularly value those two:
- > avoid polluting the current system environment with side-effects produced by Maven plugins
- > run the build from a different working directory than the current `${user.dir}`

Reply:
Ah I see - neat.  I'll have to play with that!